Cannot build `GOOS=windows GOARCH=arm64`

### Component(s)

_No response_

### What happened?

## Description

There appears to be a build-tag definition problem of some kind with `windows arm64` builds, and I think it started in v0.126.0.

## Steps to Reproduce

```shell
make genotelcontribcol
cd ./cmd/otelcontribcol
GOWORK=off GOOS=windows GOARCH=arm64 go build .
# github.com/DataDog/gohai/cpu
../../../../go/pkg/mod/github.com/!data!dog/gohai@v0.0.0-20230524154621-4316413895ee/cpu/cpu_windows.go:270:15: undefined: computeCoresAndProcessors
# github.com/DataDog/gohai/platform
../../../../go/pkg/mod/github.com/!data!dog/gohai@v0.0.0-20230524154621-4316413895ee/platform/platform_windows.go:168:29: undefined: SERVER_INFO_101
../../../../go/pkg/mod/github.com/!data!dog/gohai@v0.0.0-20230524154621-4316413895ee/platform/platform_windows.go:185:9: undefined: platGetServerInfo
# github.com/DataDog/datadog-agent/pkg/util/winutil
../../../../go/pkg/mod/github.com/!data!dog/datadog-agent/pkg/util/winutil@v0.66.0-devel.0.20250407180930-ebfcfa2817ce/winmem.go:126:15: undefined: performanceInformation
# github.com/open-telemetry/opentelemetry-collector-contrib/pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters
../../pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/pdh.go:347:89: undefined: PDH_FMT_COUNTERVALUE_DOUBLE
../../pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/pdh.go:515:77: undefined: PDH_RAW_COUNTER
../../pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go:127:12: undefined: PDH_COUNTER_INFO
../../pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go:137:12: undefined: PDH_FMT_COUNTERVALUE_DOUBLE
../../pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go:160:28: undefined: PDH_FMT_COUNTERVALUE_ITEM_DOUBLE
../../pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go:176:15: undefined: PDH_RAW_COUNTER
../../pkg/winperfcounters/internal/third_party/telegraf/win_perf_counters/performance_query.go:199:28: undefined: PDH_RAW_COUNTER_ITEM
```

## Expected Result

Successful Windows ARM64 build.

## Actual Result

Posted in repro steps.

### Collector version

`main`

### Environment information

## Environment
OS: Appears unimportant; this leverages Go cross-compile features.
Compiler(if manually compiled): `go1.24.3`


### OpenTelemetry Collector configuration

```yaml
N/A
```

### Log output

```shell
Posted in repro steps.
```

### Additional context

_No response_

--- a/.github/workflows/build-and-test-windows.yml
+++ b/.github/workflows/build-and-test-windows.yml
@@ -25,8 +25,11 @@ permissions: read-all
 
 jobs:
   setup-environment:
-    runs-on: windows-2025
-    if: ${{ github.actor != 'dependabot[bot]' && (contains(github.event.pull_request.labels.*.name, 'Run Windows') || github.event_name == 'push' || github.event_name == 'merge_group') }}
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ windows-latest, windows-11-arm ]
+    runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
       - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5
@@ -40,6 +43,32 @@ jobs:
       - name: Install Tools
         if: steps.go-setup.outputs.cache-hit != 'true'
         run: make install-tools
+
+  windows-smoke-build:
+    needs: [setup-environment]
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [ windows-latest, windows-11-arm ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
+      - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5
+        id: go-setup
+        with:
+          go-version: oldstable
+          cache-dependency-path: "**/*.sum"
+      - name: Install dependencies
+        if: steps.go-setup.outputs.cache-hit != 'true'
+        run: make -j2 gomoddownload
+      - name: Install Tools
+        if: steps.go-setup.outputs.cache-hit != 'true'
+        run: make install-tools
+      - name: Generate otelcontribcol files
+        run: make genotelcontribcol
+      - name: Build Collector
+        run: make otelcontribcol
+
   windows-unittest-matrix:
     needs: [setup-environment]
     strategy:

--- a/.github/workflows/build-and-test-windows.yml
+++ b/.github/workflows/build-and-test-windows.yml
@@ -25,7 +25,6 @@ permissions: read-all
 
 jobs:
   setup-environment:
-    runs-on: windows-2025
     if: ${{ github.actor != 'dependabot[bot]' && (contains(github.event.pull_request.labels.*.name, 'Run Windows') || github.event_name == 'push' || github.event_name == 'merge_group') }}
     strategy:
       fail-fast: true

--- a/.github/workflows/build-and-test-windows.yml
+++ b/.github/workflows/build-and-test-windows.yml
@@ -25,6 +25,8 @@ permissions: read-all
 
 jobs:
   setup-environment:
+    runs-on: windows-2025
+    if: ${{ github.actor != 'dependabot[bot]' && (contains(github.event.pull_request.labels.*.name, 'Run Windows') || github.event_name == 'push' || github.event_name == 'merge_group') }}
     strategy:
       fail-fast: true
       matrix:

